pybamm/models/submodels/porosity/reaction_driven_porosity_ode.py

```diff
@@ -0,0 +1,111 @@
+    """Reaction-driven porosity changes as an ODE
+
+    Parameters
+    ----------
+    param : parameter class
+        The parameters to use for this submodel
+    options : dict
+        Options dictionary passed from the full model
+    x_average : bool
+        Whether to use x-averaged variables (SPM, SPMe, etc) or full variables (DFN)
+
+    **Extends:** :class:`pybamm.porosity.BaseModel`
+    """
+
+    def __init__(self, param, options, x_average):
+        super().__init__(param, options)
+        self.x_average = x_average
+
+    def get_fundamental_variables(self):
+        if self.x_average is True:
+            eps_n_pc = pybamm.standard_variables.eps_n_pc
+            eps_s_pc = pybamm.standard_variables.eps_s_pc
+            eps_p_pc = pybamm.standard_variables.eps_p_pc
+
+            eps_n = pybamm.PrimaryBroadcast(eps_n_pc, "negative electrode")
+            eps_s = pybamm.PrimaryBroadcast(eps_s_pc, "separator")
+            eps_p = pybamm.PrimaryBroadcast(eps_p_pc, "positive electrode")
+        else:
+            eps_n = pybamm.standard_variables.eps_n
+            eps_s = pybamm.standard_variables.eps_s
+            eps_p = pybamm.standard_variables.eps_p
+        variables = self._get_standard_porosity_variables(eps_n, eps_s, eps_p)
+
+        return variables
+
+    def get_coupled_variables(self, variables):
+
+        if self.x_average is True:
+            j_n = variables["X-averaged negative electrode interfacial current density"]
+            j_p = variables["X-averaged positive electrode interfacial current density"]
+            # j_sei_n = variables[
+            #     "X-averaged negative electrode SEI interfacial current density"
+            # ]
+            # j_plating = variables[
+            #     "X-averaged negative electrode lithium plating "
+            #     "interfacial current density"
+            # ]
+            deps_s_dt = pybamm.PrimaryBroadcast(0, "current collector")
+        else:
+            j_n = variables["Negative electrode interfacial current density"]
+            j_p = variables["Positive electrode interfacial current density"]
+            # j_sei_n = variables["Negative electrode SEI interfacial current density"]
+            # j_plating = variables[
+            #     "Negative electrode lithium plating interfacial current density"
+            # ]
+            deps_s_dt = pybamm.FullBroadcast(
+                0, "separator", auxiliary_domains={"secondary": "current collector"}
+            )
+
+        deps_n_dt = -self.param.beta_surf_n * j_n
+        # Reaction driven porosity ODE for lithium-ion is not supported at the moment
+        # if self.options["SEI porosity change"] == "true":
+        #     beta_sei_n = self.param.beta_sei_n
+        #     deps_n_dt += beta_sei_n * j_sei_n
+        # if self.options["lithium plating porosity change"] == "true":
+        #     beta_plating = self.param.beta_plating
+        #     deps_n_dt += beta_plating * j_plating
```

Comment:
I commented it out as it is not used. Should I delete it? Because it is checking the options rather than the submodels at the moment there is no way we can enter these `if` statements.

Reply:
Yes can be removed